hip-clang.docker

```diff
@@ -67,3 +67,17 @@ RUN export HIPCC_COMPILE_FLAGS_APPEND='-O3 -Wno-format-nonliteral -parallel-jobs
 # install last released miopentensile in default, install latest commits when MIOTENSILE_VER="latest"
 RUN if [ "$MIOTENSILE_VER" = "latest" ] ; then cget -p $PREFIX install ROCmSoftwarePlatform/MIOpenTensile@7568654c938d42e9a91c6b18fb382f5b978d12fd; else cget -p $PREFIX install ROCmSoftwarePlatform/MIOpenTensile@5fe0bf4a8dc59f3ab62df929297280915372ce16; fi
 
+RUN cd ~ && \
+    export MLIR_COMMIT=06c9bca72f6cd787a98b7a2cb2ae051e0a05e862 && \
```

Comment:
[Notice] This commit does not have `MLIRMIOPEN_VERSION_FLAT` in the header. See https://github.com/ROCmSoftwarePlatform/MIOpen/pull/676#discussion_r575367256. This is not important for now, just FYI.

Reply:
Yah, thanks for pointing out. Adding a `MLIRMIOPEN_VERSION_FLAT` variable is fairly trivial, but right now we are freezing the merge to `miopen-dialect` because one developer is syncing with the llvm mainline. Once our code base is out of freezing state, I will add it in MLIR as well as updating the commit id here. By then, let's leave this unresolved.

Reply:
The latest commit include the `MLIRMIOPEN_VERSION_FLAT`. Marking this as resolved.